src/main/kotlin/uk/gov/dluhc/printapi/service/temporarycertificate/TemporaryCertificateService.kt

```diff
@@ -1,21 +1,38 @@
 package uk.gov.dluhc.printapi.service.temporarycertificate
 
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.dluhc.printapi.client.ElectoralRegistrationOfficeManagementApiClient
+import uk.gov.dluhc.printapi.database.repository.TemporaryCertificateRepository
 import uk.gov.dluhc.printapi.dto.GenerateTemporaryCertificateDto
 import uk.gov.dluhc.printapi.dto.PdfFile
 import uk.gov.dluhc.printapi.exception.GenerateTemporaryCertificateValidationException
+import uk.gov.dluhc.printapi.mapper.TemporaryCertificateMapper
 import uk.gov.dluhc.printapi.validator.service.GenerateTemporaryCertificateValidator
 
 @Service
-class TemporaryCertificateService(private val validator: GenerateTemporaryCertificateValidator) {
+class TemporaryCertificateService(
+    private val validator: GenerateTemporaryCertificateValidator,
+    private val eroClient: ElectoralRegistrationOfficeManagementApiClient,
+    private val temporaryCertificateRepository: TemporaryCertificateRepository,
+    private val temporaryCertificateMapper: TemporaryCertificateMapper,
+    private val certificatePdfTemplateDetailsFactory: CertificatePdfTemplateDetailsFactory,
+    private val pdfFactory: PdfFactory
+) {
 
     /**
      * Generates a Temporary Certificate
      * @throws GenerateTemporaryCertificateValidationException if the GenerateTemporaryCertificateDto fails validation
      */
-    fun generateTemporaryCertificate(generateTemporaryCertificateDto: GenerateTemporaryCertificateDto): PdfFile {
-        validator.validate(generateTemporaryCertificateDto)
-
-        TODO("not yet implemented")
+    @Transactional
+    fun generateTemporaryCertificate(request: GenerateTemporaryCertificateDto): PdfFile {
+        validator.validate(request)
+        val eroDetails = eroClient.getEro(request.gssCode)
+        val filename = certificatePdfTemplateDetailsFactory.getTemplateFilename(request.gssCode)
+        val temporaryCertificate = temporaryCertificateMapper.toTemporaryCertificate(request, eroDetails, filename)
+        val templateDetails = certificatePdfTemplateDetailsFactory.getTemplateDetails(temporaryCertificate)
+        temporaryCertificateRepository.save(temporaryCertificate)
```

Comment:
Minor, @Neil-Massey ,
I think it will be better to save the `temporaryCertificate` after generating the pdf contents, to prevent cases when there is an error in the pdf generation and no temporary certificate was generated, but we end up with records in our db.

Reply:
Nothing gets written to the database until the method exists cleanly - so it doesn't affect the behaviour; but will reorder for clarity.

Reply:
Yes, you're right @Neil-Massey. 

I missed the transactional annotation. 
I also agree that it will be better to re-order for clarity.

Thanks.